compiler-rt/lib/mpk_untrusted/mpk_common.h

```diff
@@ -7,11 +7,14 @@
 #include <cstdio>
 #include <cstring>
 
-#define PAGE_MPK 0
-#define SINGLE_STEP 1
+// MPK Runtime mode can be set with either of the two following flags,
+// PAGE_MPK and SINGLE_STEP_MPK. If neither are defined, default is 
+// set to SINGLE_STEP_MPK.
+#if !defined(PAGE_MPK) && !defined(SINGLE_STEP_MPK)
+  #define SINGLE_STEP_MPK
+#endif
```

Comment:
Are these completely mutually exclusive? What if they are both defined?

More typically, I would expect this to be formulated along these lines, since I think they cannot coexist

```
#ifndef PAGE_MPK
// The default behavior, i.e, single stepping
#else
// Page based MPK things
#endif
```
If the only difference is in how we handle a single call, then this logic can be localized either at the call site, or at the definition of the function.

Reply:
So there are two places where the definition changes the result, but I went ahead and changed it to the style you mentioned as it makes the approach a little more uniform and easier to read or understand at the respective places where they are used.